internal/pkg/deploy/upload/customresource/customresource_test.go

```diff
@@ -246,7 +246,61 @@ func TestBackend(t *testing.T) {
 
 	// ensure artifact paths match.
 	for _, cr := range crs {
-		require.Equal(t, wantedPaths[cr.Name()], cr.ArtifactPath())
+		require.Equal(t, fakePaths[cr.Name()], cr.ArtifactPath())
+	}
+}
+
+func TestStaticSite(t *testing.T) {
+	// GIVEN
+	fakeFS := &fakeTemplateReader{
+		files: map[string]*template.Content{
+			"custom-resources/desired-count-delegation.js": {
+				Buffer: bytes.NewBufferString("custom domain app runner"),
+			},
+			"custom-resources/alb-rule-priority-generator.js": {
+				Buffer: bytes.NewBufferString("rule priority"),
+			},
+			"custom-resources/env-controller.js": {
+				Buffer: bytes.NewBufferString("env controller"),
+			},
+		},
+	}
+	fakePaths := map[string]string{
+		"DynamicDesiredCountFunction": "manual/scripts/custom-resources/dynamicdesiredcountfunction/2611784f21e91e499306dac066aae5fd8f2ba664b38073bdd3198d2e041c076e.zip",
+		"EnvControllerFunction":       "manual/scripts/custom-resources/envcontrollerfunction/72297cacaeab3a267e371c17ea3f0235905b0da51410eb31c10f7c66ba944044.zip",
+		"RulePriorityFunction":        "manual/scripts/custom-resources/rulepriorityfunction/1385d258950a50faf4b5cd7deeecbc4bcc79a0d41d631e3977cffa0332e6f0c6.zip",
+	}
+
+	// WHEN
+	crs, err := StaticSite(fakeFS)
+
+	// THEN
+	require.NoError(t, err)
+	require.Equal(t, fakeFS.matchCount, 0, "expected path calls do not match")
+
+	actualFnNames := make([]string, len(crs))
+	for i, cr := range crs {
+		actualFnNames[i] = cr.Name()
+	}
+	require.ElementsMatch(t,
+		[]string{},
+		actualFnNames, "function names must match")
+
+	// ensure the zip files contain an index.js file.
+	for _, cr := range crs {
+		buf := new(bytes.Buffer)
+		size, err := buf.ReadFrom(cr.zipReader())
+		require.NoError(t, err)
+		r, err := zip.NewReader(bytes.NewReader(buf.Bytes()), size)
+		require.NoError(t, err)
+
+		_, err = r.Open("index.js")
+		require.NoError(t, err, "an index.js file must be present in all custom resources")
+	}
+
+	// ensure artifact paths match.
+	for _, cr := range crs {
+		require.Equal(t, fakePaths[cr.Name()], cr.ArtifactPath())
```

Comment:
What do  you think of deleting this block since we have a match count of 0 and no custom resources for static sites?

Reply:
I keep it here is because the static pattern will have a custom resource which is the step function trigger. Maybe it's ok to keep 🤔